src/api_manager/context/request_context.cc

```diff
@@ -326,6 +343,39 @@ void RequestContext::FillReportRequestInfo(
   }
 }
 
+const std::string RequestContext::FindRealClientIPAddress() {
+  auto serverConfig = service_context_->config()->server_config();
+  std::string client_real_ip_header;
+
+  if (serverConfig->has_real_client_ip_from_header_config() &&
+      serverConfig->real_client_ip_from_header_config()
+              .client_real_ip_header()
+              .length() > 0 &&
+      request_->FindHeader(kXForwardedForHeader, &client_real_ip_header)) {
+    // split headers
+    std::vector<std::string> secments;
+    split(client_real_ip_header, kClientIPHeaderDelimeter, &secments);
+    int client_real_ip_header_position =
+        serverConfig->real_client_ip_from_header_config()
+            .client_real_ip_position();
+
+    // adjust positive position
+    if (client_real_ip_header_position >= 0) {
```

Comment:
Hmm,  will it be easier if

if (pos < 0) pos = size - pos;

then make sure pos > 0 && pos < size

Reply:
Done